src/Services/QueuedJobService.php

```diff
@@ -842,27 +840,22 @@ public function runJob($jobId)
                     );
                     if (!$jobDescriptor || !$jobDescriptor->exists()) {
                         $broken = true;
-                        $this->getLogger()->error(
-                            print_r(
-                                [
-                                    'errno' => 0,
-                                    'errstr' => 'Job descriptor ' . $jobId . ' could not be found',
-                                    'errfile' => __FILE__,
-                                    'errline' => __LINE__,
-                                    'errcontext' => [],
-                                ],
-                                true
-                            ),
+                        $logger->error(
```

Comment:
Do we need to test if the $logger is not null?

Reply:
I don't think so, we also don't check it in the other dozen places. `private $logger` is a hard dependency. That's not made super clear because we tend to not inject dependencies through the constructor, but I'd say that's a different discussion (pre-existing issue in the class).

Reply:
Scratch that, just read through the diff again - I'm replacing the `Injector::inst()->get(LoggerInterface::class)` call which used to be `getLogger()`. So I'm turning an internal dependency into an external dependency, but making it implicit. I'd say that the default `$logger is not defined` error is pretty much the same than a custom `throw "dependency not met"` though? Moving this to a constructor based injection would be a cleaner API, but also an API breakage.

Reply:
Sorry, I was cranky before - I've fixed it be reinstating the old functionality, but taking preference for `$this->logger` if defined. So it should never be `null`.